src/Ajaxray/PHPWatermark/Watermark.php

```diff
@@ -212,4 +225,58 @@ private function ensureWritable(string $dirPath): void
             throw new \InvalidArgumentException("The specified destination $dirPath is not writable!");
         }
     }
+
+     /**
+     *  Prepare color for watermark command line
+     */
+    private function _prepareColorForWaterMark($color)
+    {
+        $hexColor = $this->_colorToRGB($color);
+
+        $opecity = $this->options['opacity'];
+
+        $rgba = "rgba\\(" . $hexColor[0] . "," . $hexColor[1] . "," . $hexColor[2] . "," . $opecity . "\\)";
+        return $rgba;
+    }
+    
+    private function _colorToRGB($hex)
+    {
+        $hex = strtolower($hex);
+
+        if (strpos($hex, 'rgba') !== false) {
+            preg_match('/^rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+(?:\.\d+)?))?\)$/i', $hex, $rgb);
+
+            if ($rgb) {
+                if (!empty($rgb[4]) && \XF::options()->up_wm_opecity === 0) {
+                    $this->setOpacity(intval(127 - 127 * $rgb[4]));
+                }
+
+                return [$rgb[1], $rgb[2], $rgb[3]];
+            }
+        }
+
+        if (strpos($hex, 'rgb') !== false) {
+            preg_match('/^rgb\(\s*(\d+%?)\s*,\s*(\d+%?)\s*,\s*(\d+%?)\s*\)$/i', $hex, $rgb);
+
+            if ($rgb) {
+                return [$rgb[1], $rgb[2], $rgb[3]];
+            }
+        } else {
+            $hex = str_replace('#', '', $hex);
+
+            if (utf8_strlen($hex) == 3) {
```

Comment:
`utf8_strlen` function is [not found](https://www.php.net/manual-lookup.php?pattern=utf8_strlen&amp;lang=en). Can we use `mb_strlen()` instead?

Reply:
I don't try that but  to make utf8_strlen work you should to Install php-mbstring module on server and that's will solve this issue.